Formula/lua.rb

```diff
@@ -38,16 +38,17 @@ def install
     inreplace "src/luaconf.h", "/usr/local", HOMEBREW_PREFIX
 
     # We ship our own pkg-config file as Lua no longer provide them upstream.
-    system "make", "macosx", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}"
-    system "make", "install", "INSTALL_TOP=#{prefix}", "INSTALL_MAN=#{man1}"
+    system "make", "macosx", "INSTALL_TOP=#{prefix}", "INSTALL_INC=#{include}/lua", "INSTALL_MAN=#{man1}"
+    system "make", "install", "INSTALL_TOP=#{prefix}", "INSTALL_INC=#{include}/lua", "INSTALL_MAN=#{man1}"
     (lib/"pkgconfig/lua.pc").write pc_file
 
     # Fix some software potentially hunting for different pc names.
     bin.install_symlink "lua" => "lua5.3"
     bin.install_symlink "lua" => "lua-5.3"
     bin.install_symlink "luac" => "luac5.3"
     bin.install_symlink "luac" => "luac-5.3"
-    (include/"lua5.3").install_symlink include.children
+    (include/"lua5.3").install_symlink Dir[include/"lua/*"]
+    lib.install_symlink "liblua.5.3.dylib" => "liblua5.3.dylib"
```

Comment:
Resolves a silly `vim` assumption, and sadly I'm pretty sure `vim` won't be the only ones making said assumption 🙃.

Reply:
(The assumption being that if the executable is discovered as `lua5.3` and the headers reside in `include/lua5.3` that the library must logically be called `liblua5.3` rather than `liblua.5.3`, without a fallback for checking the latter) 🙄.